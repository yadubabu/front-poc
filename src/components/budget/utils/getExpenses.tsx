import { useSelector } from "react-redux";
import { Trans } from "../../../dataTypes";
import { AppState } from "../../../redux/store";

export const GetExpenses = () => {
  const totalExpense = useSelector<AppState, number>(
    ({ account }) => account.totalExpense
  );
  const expenseBudget = useSelector<AppState, number>(
    (state) => state.budget.expenseBudget
  );
  const getRents = () => {
    return trans
      .filter((tran) => tran.type === "expense")
      .filter(
        (el) =>
          el.name.split(" ")[0] === "Rent" ||
          el.name.split(" ")[1] === "Rent" ||
          el.name.split(" ")[0] === "rent" ||
          el.name.split(" ")[1] === "rent"
      )
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const getGroceries = () => {
    return trans
      .filter((tran) => tran.type === "expense")
      .filter(
        (el) =>
          el.name === "Groceries" ||
          el.name.split(" ")[0] === "buy" ||
          el.name.split(" ")[0] === "Buy" ||
          el.name === "groceries"
      )
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const getShoppingExpense = () => {
    return trans
      .filter((tran) => tran.type === "expense")
      .filter((el) => el.name === "Shopping" || el.name === "shopping")
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const getBills = () => {
    return trans
      .filter((tran) => tran.type === "expense")
      .filter(
        (el) =>
          el.name.split(" ")[0] === "Bills" ||
          el.name.split(" ")[1] === "Bills" ||
          el.name.split(" ")[0] === "bills" ||
          el.name.split(" ")[1] === "bills"
      )
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const trans = useSelector<AppState, Trans[]>(({ trans }) => trans);
  return (
    <>
      {getRents() === 0 ? (
        ""
      ) : (
        <div className="flex text-dark align-center justify-between m-2 p-2">
          <span className="h6 w-2/3">Spent on Rents</span>
          <span className="text-xs w-1/2">
            {(getRents() * 100) / expenseBudget}% is spent
          </span>
          <span className="h6 w-1/7">{getRents()}</span>
        </div>
      )}
      {getGroceries() === 0 ? (
        ""
      ) : (
        <div className="flex text-dark align-center justify-between m-2 p-2">
          <span className="h6 w-2/3">Spent on Needs</span>
          <span className="text-xs w-1/2">
            {(getGroceries() * 100) / expenseBudget}% is spent
          </span>
          <span className="h6 w-1/7">{getGroceries()}</span>
        </div>
      )}
      {getShoppingExpense() === 0 ? (
        ""
      ) : (
        <div className="flex text-dark align-center justify-between m-2 p-2">
          <span className="h6 w-2/3">Spent on Shopping</span>
          <span className="text-xs w-1/2">
            {(getShoppingExpense() * 100) / expenseBudget}% is spent
          </span>
          <span className="h6 w-1/7">{getShoppingExpense()}</span>
        </div>
      )}
      {getBills() === 0 ? (
        ""
      ) : (
        <div className="flex text-dark align-center justify-between m-2 p-2">
          <span className="h6 w-2/3">Spent on Bills</span>
          <span className="text-xs w-1/2">
            {(getBills() * 100) / expenseBudget}% is spent
          </span>
          <span className="h6 w-1/7">{getBills()}</span>
        </div>
      )}
      {totalExpense ===
      getRents() + getGroceries() + getShoppingExpense() + getBills() ? (
        ""
      ) : (
        <div className="flex text-dark align-center justify-between m-2 p-2">
          <span className="h6 w-2/3">Spent on Others</span>
          <span className="text-xs w-1/2">
            {((totalExpense -
              getRents() -
              getShoppingExpense() -
              getGroceries() -
              getBills()) *
              100) /
              expenseBudget}
            % is spent
          </span>
          <span className="h6 w-1/7">
            {totalExpense -
              getRents() -
              getShoppingExpense() -
              getGroceries() -
              getBills()}
          </span>
        </div>
      )}
      <hr />
      <div className="save d-flex">
        <span className="h6">Total Expenses</span>
        <span className="h2 text-light">{totalExpense}</span>
      </div>
    </>
  );
};
