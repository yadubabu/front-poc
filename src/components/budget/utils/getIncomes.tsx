import React from "react";
import { useSelector } from "react-redux";
import { AppState } from "../../../redux/store";
import { Trans } from "../../../dataTypes";


const GetIncomes = () => {
  const totalIncomes = useSelector<AppState, number>(
    (state) => state.account.totalIncomes
  );
  const trans = useSelector<AppState, Trans[]>((state) => state.trans);
  const getSalaries = () => {
    return trans
      .filter((tran) => tran.type === "income")
      .filter((el) => el.name.toLowerCase() === "salary")
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const getRentals = () => {
    return trans
      .filter((tran) => tran.type === "income")
      .filter(
        (el) =>
          el.name.split(" ")[0] === "Rental" ||
          el.name.split(" ")[1] === "Rental" ||
          el.name.split(" ")[0] === "rental" ||
          el.name.split(" ")[1] === "rental"
      )
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const getSales = () => {
    return trans
      .filter((tran) => tran.type === "income")
      .filter(
        (el) =>
          el.name.split(" ")[0] === "sold" ||
          el.name.split(" ")[1] === "sold" ||
          el.name.split(" ")[0] === "Sold" ||
          el.name.split(" ")[1] === "Sold"
      )
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const getGifts = () => {
    return trans
      .filter((tran) => tran.type === "income")
      .filter(
        (el) =>
          el.name.split(" ")[0] === "gifts" ||
          el.name.split(" ")[1] === "gifts" ||
          el.name.split(" ")[0] === "Gifts" ||
          el.name.split(" ")[1] === "Gifts"
      )
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  return (
    <div className="bg-indigo-200 m-3 h-100 rounded-md">
      {getSalaries() === 0 ? (
        ""
      ) : (
        <div className="flex align-center justify-between p-3">
          <span className="h6 w-3/4">Earns by Salary</span>

          <span className="h6 w-1/9">{getSalaries()}</span>
        </div>
      )}
      {getRentals() === 0 ? (
        ""
      ) : (
        <div className="flex align-center justify-between p-3">
          <span className="h6 w-3/4">Earns by Rentals</span>
          <span className="h6 w-1/9">{getRentals()}</span>
        </div>
      )}
      {getSales() === 0 ? (
        ""
      ) : (
        <div className="flex align-center justify-between p-3">
          <span className="h6 w-3/4">Earns by Sales</span>

          <span className="h6 w-1/9">{getSales()}</span>
        </div>
      )}
      {getGifts() === 0 ? (
        ""
      ) : (
        <div className="flex align-center justify-between p-3">
          <span className="h6 w-3/4">Earns by Gifts</span>
          <span className="h6 w-1/9">{getGifts()}</span>
        </div>
      )}
      {totalIncomes -
        getGifts() -
        getRentals() -
        getSalaries() -
        getSales() ===
      0 ? (
        ""
      ) : (
        <div className="flex align-center justify-between p-3">
          <span className="h6 w-3/4">Earns by Others</span>
          <span className="h6 w-1/9">
            {totalIncomes -
              getGifts() -
              getRentals() -
              getSalaries() -
              getSales()}
          </span>
        </div>
      )}
      <hr />
      <div className="flex  align-center justify-between p-3">
        <span className="h6 w-3/4 text-md font-bold">Total Earnings</span>

        <span className="h6 w-1/9 text-md font-bold">{totalIncomes}</span>
      </div>{" "}
    </div>
  );
};

export default GetIncomes;
