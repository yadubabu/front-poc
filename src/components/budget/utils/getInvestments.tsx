import { useSelector } from "react-redux";
import { Trans } from "../../../dataTypes";
import { AppState } from "../../../redux/store";

export const GetInvestment = () => {
  const totalInvestment = useSelector<AppState, number>(
    ({ account }) => account.totalInvestment
  );
  const investmentBudget = useSelector<AppState, number>(
    (state) => state.budget.investmentBudget
  );

  const getGoldInvestments = () => {
    return trans
      .filter((tran) => tran.type === "investment")
      .filter(
        (el) =>
          el.name.split(" ")[1] === "Gold" || el.name.split(" ")[1] === "gold"
      )
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const getStocks = () => {
    return trans
      .filter((tran) => tran.type === "investment")
      .filter(
        (el) =>
          el.name.split(" ")[1] === "stocks" ||
          el.name.split(" ")[1] === "Stocks"
      )
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const getLandInvestment = () => {
    return trans
      .filter((tran) => tran.type === "investment")
      .filter(
        (el) =>
          el.name.split(" ")[1] === "land" || el.name.split(" ")[1] === "Land"
      )
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const trans = useSelector<AppState, Trans[]>(({ trans }) => trans);
  return (
    <>
      {getGoldInvestments() === 0 ? (
        ""
      ) : (
        <div className="flex text-dark align-center justify-between m-2 p-2">
          <span className="h6 w-2/3">Spent on Gold</span>
          <span className="text-xs w-1/2">
            {Math.round((getGoldInvestments() * 100) / investmentBudget)}% is
            spent
          </span>
          <span className="h6 w-1/7">{getGoldInvestments()}</span>
        </div>
      )}
      {getLandInvestment() === 0 ? (
        ""
      ) : (
        <div className="flex text-dark align-center justify-between m-2 p-2">
          <span className="h6 w-2/3">Spent on Land</span>
          <span className="text-xs w-1/2">
            {Math.round((getLandInvestment() * 100) / investmentBudget)}% is
            spent
          </span>
          <span className="h6 w-1/7">{getLandInvestment()}</span>
        </div>
      )}
      {getStocks() === 0 ? (
        ""
      ) : (
        <div className="flex text-dark align-center justify-between m-2 p-2">
          <span className="h6 w-2/3">Spent on Stocks</span>
          <span className="text-xs w-1/2">
            {Math.round((getStocks() * 100) / investmentBudget)}% is spent
          </span>
          <span className="h6 w-1/7">{getStocks()}</span>
        </div>
      )}
      {totalInvestment ===
      getGoldInvestments() + getLandInvestment() + getStocks() ? (
        ""
      ) : (
        <div className="flex text-dark align-center justify-between m-2 p-2">
          <span className="h6 w-2/3">Investments on Others </span>
          <span className="text-xs w-1/2">
            {Math.round(
              ((totalInvestment -
                getGoldInvestments() -
                getStocks() -
                getLandInvestment()) *
                100) /
                investmentBudget
            )}
            % is spent
          </span>
          <span className="text-xs w-1/7">
            {totalInvestment -
              getGoldInvestments() -
              getLandInvestment() -
              getStocks()}
          </span>
        </div>
      )}
      <hr />
      <div className="flex text-dark align-center justify-between m-2 p-2">
          <span className="h5 w-2/3">Total Investments</span>
          <span className="h5 text-sm w-1/2">
          ({Math.round((totalInvestment * 100) / investmentBudget)}% total
          spent)
        </span>

        <span className="h5">{totalInvestment}</span>
      </div>
    </>
  );
};
