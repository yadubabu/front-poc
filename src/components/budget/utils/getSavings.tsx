import { useSelector } from "react-redux";
import { Trans } from "../../../dataTypes";
import { AppState } from "../../../redux/store";

export const GetSavings = () => {
  const trans = useSelector<AppState, Trans[]>(({ trans }) => trans);
  const savingsBudget = useSelector<AppState, number>(
    (state) => state.budget.savingsBudget
  );
  const totalSavings = useSelector<AppState, number>(
    ({ account }) => account.totalSavings
  );
  const getBonds = () => {
    return trans
      .filter((tran) => tran.type === "savings")
      .filter(
        (el) =>
          el.name.split(" ")[1] === "Bonds" || el.name.split(" ")[1] === "bonds"
      )
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const getDeposites = () => {
    return trans
      .filter((tran) => tran.type === "savings")
      .filter(
        (el) =>
          el.name.split(" ")[0] === "Deposit" ||
          el.name.split(" ")[0] === "deposit"
      )
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const getFunds = () => {
    const funds = trans
      .filter((tran) => tran.type === "savings")
      .filter(
        (el) =>
          el.name.split(" ")[1] === "fund" || el.name.split(" ")[1] === "Fund"
      )
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
    if (funds) {
      return funds;
    } else {
      return 0;
    }
  };

  return (
    <>
      {getBonds() === 0 ? (
        ""
      ) : (
        <div className="flex text-dark align-center justify-between m-2 p-2">
          <span className="h6 w-2/3">Spent on Bonds</span>
          <span className="text-xs w-1/2">
            {Math.round((getBonds() * 100) / savingsBudget)}% is spent
          </span>
          <span className="h6 w-1/7">{getBonds()}</span>
        </div>
      )}
      {getDeposites() === 0 ? (
        ""
      ) : (
        <div className="flex text-dark align-center justify-between m-2 p-2">
          <span className="h6 w-2/3">Spent on Deposits</span>
          <span className="text-xs w-1/2">
            {Math.round((getDeposites() * 100) / savingsBudget)}% is spent
          </span>
          <span className="h6 w-1/7">{getDeposites()}</span>
        </div>
      )}
      {getFunds() === 0 ? (
        ""
      ) : (
        <>
          <div className="flex text-dark align-center justify-between m-2 p-2">
            <span className="h6 w-2/3">Spent on Emergency Funds</span>
            <span className="text-xs w-1/2">
              {Math.round((getFunds() * 100) / savingsBudget)}% is spent
            </span>
            <span className="h6 w-1/7">{getFunds()}</span>
          </div>
        </>
      )}
      {totalSavings === getBonds() + getDeposites() + getFunds() ? (
        ""
      ) : (
        <div className="flex text-dark m-2 p-2">
          <span className="h6 w-2/3">Spent on Other Savings</span>
          <span className="text-xs w-1/2">
            {Math.round(
              ((totalSavings - getDeposites() - getFunds() - getBonds()) *
                100) /
                savingsBudget
            )}
            % is spent
          </span>
          <span className="h6 w-1/7">
            {totalSavings - getBonds() - getFunds() - getDeposites()}
          </span>
        </div>
      )}
      <hr />
      <div className="flex text-dark align-center justify-between m-2 p-2">
        <span className="h5 w-2/3">Total Savings</span>
        <span className="h5 text-xs w-1/2 font-bold">
          ({Math.round((totalSavings * 100) / savingsBudget)}% is spent)
        </span>
        <span className="h5">{totalSavings}</span>
      </div>
    </>
  );
};
