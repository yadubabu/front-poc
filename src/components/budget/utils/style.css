.save {
  display: flex;
  align-items: center;
  justify-content: space-between;
}
.exp {
  display: flex;
  align-items: center;
  justify-content: space-around;
}
.tot {
  padding: 0 100px;
}
@media (max-width: 786px) {
  .tot {
    padding: 0 10px;
  }
}
