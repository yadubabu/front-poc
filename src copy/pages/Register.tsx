import React, { useState } from "react";
import Col from "react-bootstrap/Col";
import Form from "react-bootstrap/Form";
import Row from "react-bootstrap/Row";
import { FieldValues, useForm } from "react-hook-form";
import axios from "axios";
import { registerApi } from "../redux/apis";
import { useSelector } from "react-redux";
import { Auth } from "../dataTypes";
import logImg from "../assets/home1.jpg";
import "./style.css";
import { registerOptions } from "../validators/register/registerOptions";
import RegisterErrors from "../validators/register/RegisterErrors";

type FormData = {
  name: string;
  email: string;
  password: string;
  confirmpassword: string;
  pancard: string;
  phone: number;
};

const Register = () => {
  const auth = useSelector<Auth>((state) => state.auth);
  const [msg, setMsg] = useState("");
  const {
    register,
    handleSubmit,
    reset,
    formState: { errors },
  } = useForm<FormData>({ mode: "onChange" });
  const submitHandle = async (data: FieldValues) => {
    const { name, email, password, confirmpassword, pancard, phone } = data;

    const response = await axios.post(`${registerApi}`, {
      name,
      email,
      password,
      confirmpassword,
      pancard,
      phone,
    });
    if (response) {
      setMsg("Successfully Registered");
      reset();
    }
  };

  return (
    <>
      <div className="logImg">
        <img src={logImg} alt="" />
      </div>
      <div className="register">
        {msg === "Successfully Registered" ? (
          <>{(window.location.href = "/login")}</>
        ) : (
          ""
        )}
        {!auth && (
          <>
            <Form className="p-4" onSubmit={handleSubmit(submitHandle)}>
              <Form.Group
                as={Row}
                className="mb-3 m-2 p-2"
                controlId="formPlaintextName"
              >
                <Form.Label column sm="4">
                  Name
                </Form.Label>
                <Col sm="8">
                  <Form.Control
                    {...register("name", registerOptions.name)}
                    type="text"
                    placeholder="Enter your name"
                  />
                  <small className="h6 text-danger">
                    {errors.name && (
                      <RegisterErrors msg={errors.name.message} />
                    )}
                  </small>{" "}
                </Col>
              </Form.Group>
              <Form.Group
                as={Row}
                className="mb-3 m-2 p-2"
                controlId="formPlaintextEmail"
              >
                <Form.Label column sm="4">
                  Email
                </Form.Label>
                <Col sm="8">
                  <Form.Control
                    {...register("email", registerOptions.email)}
                    type="text"
                    placeholder="Enter your Email"
                  />
                  <small className="h6 text-danger">
                    {errors.email && (
                      <RegisterErrors msg={errors.email.message} />
                    )}
                  </small>{" "}
                </Col>
              </Form.Group>
              <Form.Group
                as={Row}
                className="mb-3 m-2 p-2"
                controlId="formPlaintextPassword"
              >
                <Form.Label column sm="4">
                  Password
                </Form.Label>
                <Col sm="8">
                  <Form.Control
                    {...register("password", registerOptions.password)}
                    type="password"
                    placeholder="Enter your password"
                  />
                  <small className="h6 text-danger">
                    {errors.password && (
                      <RegisterErrors msg={errors.password.message} />
                    )}
                  </small>{" "}
                </Col>
              </Form.Group>
              <Form.Group
                as={Row}
                className="mb-3 m-2 p-2"
                controlId="formPlaintextPassword"
              >
                <Form.Label column sm="4">
                  Confirm Password
                </Form.Label>
                <Col sm="8">
                  <Form.Control
                    {...register(
                      "confirmpassword",
                      registerOptions.confirmpassword
                    )}
                    type="password"
                    placeholder="Enter your confirm password"
                  />
                  <small className="h6 text-danger">
                    {errors.confirmpassword && (
                      <RegisterErrors msg={errors.confirmpassword.message} />
                    )}
                  </small>{" "}
                </Col>
              </Form.Group>
              <Form.Group
                as={Row}
                className="mb-3 m-2 p-2"
                controlId="formPlaintextPancard"
              >
                <Form.Label column sm="4">
                  Pancard
                </Form.Label>
                <Col sm="8">
                  <Form.Control
                    {...register("pancard", registerOptions.pancard)}
                    type="text"
                    placeholder="Enter your Pancard nunber"
                  />
                  <small className="h6 text-danger">
                    {errors.pancard && (
                      <RegisterErrors msg={errors.pancard.message} />
                    )}
                  </small>{" "}
                </Col>
              </Form.Group>
              <Form.Group
                as={Row}
                className="mb-3 m-2 p-2"
                controlId="formPlainNumberPhone"
              >
                <Form.Label column sm="4">
                  Phone
                </Form.Label>
                <Col sm="8">
                  <Form.Control
                    {...register("phone", registerOptions.phone)}
                    type="number"
                    placeholder="Enter your Phone"
                  />
                  <small className="h6 text-danger">
                    {errors.phone && (
                      <RegisterErrors msg={errors.phone.message} />
                    )}
                  </small>{" "}
                </Col>
              </Form.Group>
              <Form.Group
                as={Row}
                className="mb-3 m-2 p-2"
                controlId="formPlaintextPassword"
              >
                <Col sm="4">
                  <Form.Control
                    className="logBtn btn btn-success"
                    type="submit"
                    value="Register"
                  />
                </Col>
              </Form.Group>
            </Form>{" "}
          </>
        )}
      </div>
    </>
  );
};

export default Register;
