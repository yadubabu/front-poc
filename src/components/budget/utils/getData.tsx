import { useSelector } from "react-redux";
import { Trans } from "../../../dataTypes";
import { AppState } from "../../../redux/store";
import "./style.css";
import salImg from "../../../assets/salary.png";
import { BsMotherboardFill } from "react-icons/bs";
import { AiFillGold } from "react-icons/ai";
import { GiIsland } from "react-icons/gi";

export const GetSavings = () => {
  const trans = useSelector<AppState, Trans[]>(({ trans }) => trans);
  const totalSavings = useSelector<AppState, number>(
    ({ account }) => account.totalSavings
  );
  const totalExpenses = useSelector<AppState, number>(
    ({ account }) => account.totalSavings
  );
  const totalInvestments = useSelector<AppState, number>(
    ({ account }) => account.totalSavings
  );
  const getSalary = () => {
    return trans
      .filter((tran) => tran.type === "savings")
      .filter((el) => el.name === "Salary")
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const getSoldIncomes = () => {
    const sold = trans
      .filter((tran) => tran.type === "savings")
      .filter((el) => el.name === "Sold car")
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
    if (sold) {
      return sold;
    } else {
      return 0;
    }
  };

  return (
    <>
      {getSalary() === 0 ? (
        ""
      ) : (
        <div className="save d-flex">
          <span className="h6 text-light">
            Earnings By Salary{" "}
            <img src={salImg} alt="" width={50} height={50} />
          </span>
          <span className="h2 text-light">{getSalary()}</span>
        </div>
      )}
      {getSoldIncomes() === 0 ? (
        ""
      ) : (
        <>
          <div className="save d-flex">
            <span className="h6 text-light">Earnings By Soldings </span>
            <span className="h2 text-light">{getSoldIncomes()}</span>
          </div>
        </>
      )}
      {totalSavings === getSalary() + getSoldIncomes() ? (
        ""
      ) : (
        <div className="save d-flex">
          <span className="h6 text-light">
            Earnings By Others{" "}
            <span className="text-success">
              {" "}
              <BsMotherboardFill />{" "}
            </span>
          </span>

          <span className="h2 text-light">
            {totalSavings - getSalary() - getSoldIncomes()}
          </span>
        </div>
      )}
      <hr />
      <div className="save d-flex">
        <span className="h6 text-light">Total Savings</span>
        <span className="h2 text-light">{totalSavings}</span>
      </div>
    </>
  );
};
export const GetExpenses = () => {
  const getTotalExpenses = () => {
    return trans
      .filter((tran) => tran.type === "expense")
      .map((el) => el.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const getRents = () => {
    return trans
      .filter((tran) => tran.type === "expense")
      .filter((el) => el.name === "Rent")
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const getGroceries = () => {
    return trans
      .filter((tran) => tran.type === "expense")
      .filter((el) => el.name === "Groceries")
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const getShoppingExpense = () => {
    return trans
      .filter((tran) => tran.type === "expense")
      .filter((el) => el.name === "Shopping")
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const getOtherExpenses = () => {
    return trans
      .filter((tran) => tran.type === "expense")
      .filter((el) => el.name === "Others")
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const trans = useSelector<AppState, Trans[]>(({ trans }) => trans);
  return (
    <>
      <div className="save d-flex">
        <span className="h6">Expense on Rents </span>
        <span className="h6 text-light">{getRents()}</span>
      </div>
      <div className="save d-flex">
        <span className="h6">Expense on Groceries </span>
        <span className="h6 text-light">{getGroceries()}</span>
      </div>
      <div className="save d-flex">
        <span className="h6">Expense on Shoppings </span>
        <span className="h6 text-light">{getShoppingExpense()}</span>
      </div>
      <div className="save d-flex">
        <span className="h6">Expense on Others </span>
        <span className="h6 text-light">{getOtherExpenses()}</span>
      </div>

      <hr />
      <div className="save d-flex">
        <span className="h6">Total Savings</span>
        <span className="h2 text-light">{getTotalExpenses()}</span>
      </div>
    </>
  );
};

export const GetInvestment = () => {
  const getTotalInvestments = () => {
    return trans
      .filter((tran) => tran.type === "investment")
      .map((el) => el.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const getGoldInvestments = () => {
    return trans
      .filter((tran) => tran.type === "investment")
      .filter((el) => el.name === "Buy Gold")
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const getLandInvestment = () => {
    return trans
      .filter((tran) => tran.type === "investment")
      .filter((el) => el.name === "Buy Land")
      .map((e) => e.amount)
      .reduce((a: number, c: number) => {
        return a + c;
      }, 0);
  };
  const trans = useSelector<AppState, Trans[]>(({ trans }) => trans);
  return (
    <>
      <div className="save d-flex">
        <span className="h6">
          Investments on Gold{" "}
          <span className="text-warning">
            <AiFillGold />
          </span>
        </span>

        <span className="h6 text-light">{getGoldInvestments()}</span>
      </div>
      <div className="save d-flex">
        <span className="h6">
          Investments on Land
          <span className="text-danger">
            <GiIsland />
          </span>{" "}
        </span>
        <span className="h6 text-light">{getLandInvestment()}</span>
      </div>
      <div className="save d-flex">
        <span className="h6">
          Expense on Others{" "}
          <span className="text-primary">
            <BsMotherboardFill />
          </span>
        </span>
        <span className="h6 text-light">
          {getTotalInvestments() - getGoldInvestments() - getLandInvestment()}
        </span>
      </div>
      <hr />
      <div className="save d-flex">
        <span className="h6 ">Total Investments</span>
        <span className="h2 text-light">{getTotalInvestments()}</span>
      </div>
    </>
  );
};
