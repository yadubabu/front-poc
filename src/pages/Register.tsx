import React, { useState } from "react";
import Col from "react-bootstrap/Col";
import Form from "react-bootstrap/Form";
import Row from "react-bootstrap/Row";
import { FieldValues, useForm } from "react-hook-form";
import axios from "axios";
import { registerApi } from "../redux/apis";
import { useSelector } from "react-redux";
import { Auth } from "../dataTypes";
import logImg from "../assets/home1.jpg";
import "./style.css";
import { registerOptions } from "../validators/register/registerOptions";
import RegisterErrors from "../validators/register/RegisterErrors";
import LoadingPage from "./LoadingPage";
import MessageModal from "../components/MessageModal";
type FormData = {
  name: string;
  email: string;
  password: string;
  confirmpassword: string;
  pancard: string;
  phone: number;
};

const Register = () => {
  const [showModal, setShowModal] = useState(false);
  const auth = useSelector<Auth>((state) => state.auth);
  const [msg, setMsg] = useState("");

  const {
    register,
    handleSubmit,
    reset,
    formState: { errors },
  } = useForm<FormData>({ mode: "onSubmit" });
  const submitHandle = async (data: FieldValues) => {
    const { name, email, password, confirmpassword, pancard, phone } = data;

    const response = await axios.post(`${registerApi}`, {
      name,
      email,
      password,
      confirmpassword,
      pancard,
      phone,
    });
    if (response) {
      setMsg("Successfully Registered");
      setShowModal(true);
      reset();
    }
  };

  return (
    <>
      {!auth && (
        <>
          <form
            className="form "
            id="regi"
            onSubmit={handleSubmit(submitHandle)}
          >
            {msg === "Successfully Registered" ? (
              <>
                <MessageModal msg={msg} />
              </>
            ) : (
              ""
            )}
            <div className="row m-4">
              <label className="col-4 m-1">Name</label>
              <input
                className="col-7 m-1"
                {...register("name", registerOptions.name)}
                type="text"
                placeholder="Enter your Name"
              />
              <small className="text-danger">
                {errors.name && <RegisterErrors msg={errors.name.message} />}
              </small>
            </div>
            <div className="row m-4">
              <label className="col-4 m-1">Email</label>
              <input
                className="col-7 m-1"
                {...register("email", registerOptions.email)}
                type="text"
                placeholder="Enter your Email"
              />
              <small className="text-danger">
                {errors.email && <RegisterErrors msg={errors.email.message} />}
              </small>
            </div>
            <div className="row m-4">
              <label className="col-4 m-1">Password</label>
              <input
                className="col-7 m-1"
                {...register("password", registerOptions.password)}
                type="password"
                placeholder="Enter your password"
              />
              <small className="text-danger">
                {errors.password && (
                  <RegisterErrors msg={errors.password.message} />
                )}
              </small>
            </div>
            <div className="row m-4">
              <label className="col-4 m-1">ConfirmPassword</label>
              <input
                className="col-7 m-1"
                {...register(
                  "confirmpassword",
                  registerOptions.confirmpassword
                )}
                type="password"
                placeholder="Retype your password"
              />
              <small className="text-danger">
                {errors.confirmpassword && (
                  <RegisterErrors msg={errors.confirmpassword.message} />
                )}
              </small>
            </div>
            <div className="row m-4">
              <label className="col-4 m-1">Pancard</label>
              <input
                className="col-7 m-1"
                {...register("pancard", registerOptions.pancard)}
                type="text"
                placeholder="Enter your Pancard"
              />
              <small className="text-danger">
                {errors.pancard && (
                  <RegisterErrors msg={errors.pancard.message} />
                )}
              </small>
            </div>
            <div className="row m-4">
              <label className="col-4 m-1">Phone</label>
              <input
                className="col-7 m-1"
                {...register("phone", registerOptions.phone)}
                type="number"
                placeholder="Enter your phone number"
              />
              <small className="text-danger mx-10">
                {errors.phone && <RegisterErrors msg={errors.phone.message} />}
              </small>
            </div>
            <div className="row col-3 m-4 ">
              <input
                className="formBtn btn btn-warning center"
                type="submit"
                value="Register"
              />
            </div>
          </form>
        </>
      )}
    </>
  );
};

export default Register;
