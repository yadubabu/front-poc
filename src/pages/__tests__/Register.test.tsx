import renderer from 'react-test-renderer';
import { onlywrap } from '../../utils/test_helper';
import Register from '../Register';


describe('Register Page Testing',()=>{
    test('SnapShot testing....',()=>{
        const wrapper=renderer.create(onlywrap(<Register />)).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
})