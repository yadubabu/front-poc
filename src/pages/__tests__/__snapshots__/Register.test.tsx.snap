// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Page Testing SnapShot testing.... 1`] = `null`;
